### DISCUSSION TITLE
Common Installation Errors

### DISCUSSION THREAD

**Discussion by Aptronymist** [vladmandic/sdnext#1627] [General]
We're keeping track of our most common installation errors here.

For starters there is a new upstream package conflict with `typing-extensions`

If you encounter an error related to `typing-extensions` on first running SD.Next it is due to a very recent conflict in upstream Python libraries.  
*This is a known issue that we can't do anything about.*  
Rest assured, there is a simple resolution for this issue, please do the following:

Run `webui.bat` or `webui.sh` again. This will trigger the automatic resolution process.
Upon re-execution, SD.Next will resolve the conflict and function as intended from then on.

**Comment by Aptronymist** [comment]
Anyone experiencing problems with clip-interrogator, please be aware that there is a known issue with it, as the switch to diffusers required a new transformers package, which the old clip-interrogator is not compatible with. 
Be aware that clip-interrogator is built-in to SD.Next, so if you have installed it as an extension yourself, you will need to remove it.

SD.Next now includes a different/newer version of clip-interrogator for new installs, but it doesn't uninstall existing one to prevent any issues on a users system.

You can try changing existing clip-interrogator directly using the following command:

>git submodule set-url extensions-builtin/clip-interrogator-ext [Dahvikiin/clip-interrogator-ext.git](https://github.com/Dahvikiin/clip-interrogator-ext.git)

Then run `webui.bat/webui.sh --upgrade` and you should be good to go!

**Comment by Aptronymist** [comment]
For users experiencing a `'StableDiffusionXLPipeline' object has no attribute 'decode_first_stage'` error, the issue is with the Image Previews.
Make sure your chosen method is *NOT* `Full VAE`, the others should be working, if you continue to have errors, disable previews and see if that resolves it.

**Comment by vladmandic** [comment]
If you experience

> 14:56:18-652133 ERROR Diffuser model downloaded error: model=stabilityai/stable-diffusion-xl-base-0.9 module 'diffusers' has no attribute 'StableDiffusionXLPipeline'

that means that some extension you have installed has downgraded version of the required `diffusers` package. disable user extensions.

**Comment by Aptronymist** [comment]
If you are getting errors with xyz_grid.py, or `AttributeError: 'Options' object has no attribute 'uni_pc_order'`, you need to run SD.Next without any `--backend` arguments to initialize some legacy variables that are not initialized if you start with `--backend diffusers` or `--backend original` yet.
It should be corrected soon as to no longer be an issue.
